### DISCUSSION TITLE
Custom edge processor

### DISCUSSION THREAD

**Discussion by ehardesty** [SixLabors/ImageSharp#1297] [Q&A] [answered]
I need to detect edges but only want horizontal or vertical images but not both in the same image.
I had planned to use Edge2DDetectorProcessor with my own modified Sobel kernels but everything appears to be internal. 

Am I missing something or will I need to reimplement a lot of the infrastructure for what could be a pretty simple change. I would prefer not to reinvent the wheel here just to isolate the X and Y kernels.

**Comment by JimBobSquarePants** [comment]
Yeah, I think the API can be better here, it's not extensible at all.

I think dropping the enum and creating methods for each edge detection type would be much better. It'll break binary compatibility but I think it's an important enough change to do so between the RC and V1.

**Reply by tocsoft** [reply]
@JimBobSquarePants we don't have to break compatibility we should be able support both with little difficulty so we could put this off post v1 if we wanted to.

**Reply by JimBobSquarePants** [reply]
@tocsoft The current design really isn't great. I'll open a PR tonight and show you what my replacement will be.

**Comment by JimBobSquarePants** [comment, marked as answer]
@dalingrin I've just raised #1299 which will allow you to do what you wish. I'd really appreciate it if you could cast your eye over the new API to ensure it meets your requirements.
